src/index.ts

```diff
@@ -1,53 +1,92 @@
-import { AnyLayer, IControl, Map } from 'maplibre-gl';
+import {ControlPosition, IControl, LayerSpecification, Map} from 'maplibre-gl';
 
-import { play, pause, reload, skipBackward, skipForward } from './icons';
+import {pause, play, reload, skipBackward, skipForward} from './icons';
 
 const ACTIVE_BUTTON_COLOR = 'rgb(204, 204, 204)';
 
 type ContainerOptions = {
-    length: number;
-    interval: number;
-    onSliderValueChange: () => void;
+  length: number;
+  interval: number;
+  loopDelay: number;
+  showButtons?: boolean;
+  onSliderValueChange: () => void;
+  loop?: boolean;
+  autoplay?: boolean;
```

Comment:
IMHO,  these options should not be optional because these are not exposed and therefore we can set default values.

```suggestion
  length: number;
  interval: number;
  loopDelay: number;
  showButtons: boolean;
  onSliderValueChange: () => void;
  loop: boolean;
  autoplay: boolean;
```